app/models/miq_action.rb

```diff
@@ -200,6 +200,38 @@ def invoke_action_for_built_in_policy(rec, inputs)
     send(method.to_sym, self, rec, inputs)
   end
 
+  def invoke_or_queue(
+    synchronous,
+    calling_method,
+    role,
+    zone,
+    target,
+    target_method,
+    target_args = [],
+    log_suffix = nil
+  )
```

Comment:
@moolitayer I am ok with the list of arguments for now as it is clear with needs to be passed.  I would ask that we make the `invoke_or_queue` method private then I think we can merge this.